server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -92,6 +104,8 @@ public class RemoteFsTranslog extends Translog {
     private final Semaphore syncPermit = new Semaphore(SYNC_PERMIT);
     private final AtomicBoolean pauseSync = new AtomicBoolean(false);
     private final boolean isTranslogMetadataEnabled;
+    private final Map<Long, String> metadataFilePinnedTimestampMap;
+    private final Map<String, Tuple<Long, Long>> metadataFileGenerationMap;
```

Comment:
This cache is just for the metadata files and corresponding generations that are pinned ? if yes, adding `pinned` in the naming might increase code readability